pkg/engine/circleci.go

```diff
@@ -27,17 +26,13 @@ func (e *Engine) ScanCircleCI(ctx context.Context, token string) error {
 		return err
 	}
 
-	handle, err := e.sourceManager.Enroll(ctx, "trufflehog - Circle CI", new(circleci.Source).Type(),
-		func(ctx context.Context, jobID, sourceID int64) (sources.Source, error) {
-			circleSource := circleci.Source{}
-			if err := circleSource.Init(ctx, "trufflehog - Circle CI", jobID, sourceID, true, &conn, runtime.NumCPU()); err != nil {
-				return nil, err
-			}
-			return &circleSource, nil
-		})
-	if err != nil {
+	sourceName := "trufflehog - Circle CI"
+	sourceID, jobID, _ := e.sourceManager.GetIDs(ctx, sourceName, new(circleci.Source).Type())
+
+	circleSource := &circleci.Source{}
```

Comment:
this is definitely just a weird little nit but if you're going to create a new source anyway, why not use that one to get a type to pass to `GetIDs`?

Reply:
also, is discarding the error here the right thing to do?

Reply:
yeah, it's definitely weird. I have it this way because I only want to get the source type to pass to `GetIDs`. I'll make a followup PR to add constants to each source and use that instead of creating an object just to call `Type()`.

> also, is discarding the error here the right thing to do?

For now, the engine always uses the default API implementation (i.e. no API) which never fails. I could add error checks in case that ever changes though.

Reply:
followup PR: https://github.com/trufflesecurity/trufflehog/pull/1768